azure/converters/spotinstances.go

```diff
@@ -45,36 +44,9 @@ func GetSpotVMOptionsV2(spotVMOptions *infrav1.SpotVMOptions, diffDiskSettings *
 
 	// Set the spot vm eviction policy if provided.
 	var evictionPolicy *armcompute.VirtualMachineEvictionPolicyTypes
-	if spotVMOptions.EvictionPolicy != nil {
+	if ptr.Deref(spotVMOptions.EvictionPolicy, "") != "" {
```

Comment:
Is it possible we'd ever land here with a non-nil `EvictionPolicy` with a value of `""`? It looks like this would behave differently in that case.